flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/ExecutionCheckpointingOptions.java

```diff
@@ -196,4 +196,21 @@
                                     .text(
                                             "Forces unaligned checkpoints, particularly allowing them for iterative jobs.")
                                     .build());
+
+    public static final ConfigOption<Long> CHECKPOINT_ID_OF_IGNORED_IN_FLIGHT_DATA =
+            ConfigOptions.key("execution.checkpointing.id-of-ignored-in-flight-data")
```

Comment:
It's not clear to me just from the name that it only applies to recovery.
How about
- `execution.checkpointing.recover-without-channel-state.checkpoint-id`?
- `execution.checkpointing.unaligned.allow-unsafe-recovery-from-checkpoint-id`?

Reply:
`execution.checkpointing.recover-without-channel-state.checkpoint-id` looks good to me